### PR TITLE
The Donor Dashboard are now only be generated in admin-side pages

### LINKED ISSUE #5767: The Donor Dashboard should only be generated in admin-side pages
## User Story

As an admin, I want the generate donor dashboard method to only be called when necessary so that it cannot be triggered by loading frontend pages.

## Details

This bug was reported via the Free Support and points out that the Donor Dashboard page is being generated on non-admin pages, which is being blocked by certain security measures. 

## Expected Behavior

The migration should not be triggerable by front-end pages at all.## Steps to Reproduce

## Additional Context

Original post: https://wordpress.org/support/topic/generatedonordashboardpage-and-give_after_install-always-called/

## Acceptance Criteria

-   [ ] generateDonorDashboard method is only called when necessary (not on every page load)

### PR DESCRIPTION
**Pull request by henryholtgeerts**
Resolves #5767 

## Description

This PR ensures dashboard pages are only generated by users with the correct capabilities.

## Affects

This PR affects the request handler class for Donor Dashboards.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/DonorDashboards/RequestHandler.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ public function filterQueryVars( $vars ) {
 	 */
 	public function parseRequest( $query ) {
 
-		if ( is_admin() && array_key_exists( 'give-generate-donor-dashboard-page', $query->query_vars ) ) {
+		if ( is_admin() && current_user_can( 'manage_options' ) && array_key_exists( 'give-generate-donor-dashboard-page', $query->query_vars ) ) {
```
Comment (JasonTheAdams) [inline src/DonorDashboards/RequestHandler.php:35]: Let's add `!wp_doing_ajax()` to rule out AJAX requests. I'd like this to only happen for admins requesting a page admin-side.

Comment (henryholtgeerts) [reply src/DonorDashboards/RequestHandler.php]: Done, ref: https://github.com/impress-org/givewp/pull/5768/commits/b500e825514b3efb1bf20c68a1dcdc0791bd974d

**Review by JasonTheAdams — APPROVED** [review]
Honestly, I wasn't 100% sure _how_ to test this. I made a fresh install using GiveWP 2.9.7 and then updated to 2.10.0 and followed the steps. I checked to make sure it wasn't ever created until then. So... it all seems to be working, but I'm not confident I tested the specific bug. That said, the code change is straight forward enough to make me feel pretty good.

**Comment by JasonTheAdams** [comment]
@henryholtgeerts Let me know if you feel this is tested properly. If you feel it's missing something please test it further. Otherwise, go ahead and merge this.

**Comment by henryholtgeerts** [comment]
@JasonTheAdams I feel confident in the testing of it, and checked that it doesn't break any functionality (tested by trashing my donor dashboard page and generating a new one, which invokes this method).

**Pull request merged**